jcl/src/java.base/share/classes/java/lang/invoke/MethodHandle.java

```diff
@@ -872,6 +872,20 @@ public MethodHandle bindTo(Object value) throws IllegalArgumentException, ClassC
 		return MethodHandles.insertArguments(this, 0, value);
 	}
 
+
+	boolean isUnnecesssaryPermute(MethodType permuteType, int... permute) {
+		if (this.type != permuteType) {
+			return false;
+		}
+
+		for (int i = 0; i < permute.length; i++) {
+			if (permute[i] != i) {
+				return false;
+			}
+		}
```

Comment:
Good catch here that the MTs are not sufficient to determine these are the same.  Can you add a javadoc comment to this function that describes the algorithm and why it's necessary to do both?

Reply:
Javadoc added.